concordium-contracts-common/src/schema.rs

```diff
@@ -101,6 +110,45 @@ pub struct ContractV2 {
     pub receive: BTreeMap<String, FunctionV2>,
 }
 
+/// Describes all the schemas of a V1 smart contract with a V3 schema.
+#[derive(Debug, Default, Clone, PartialEq, Eq)]
+/// The [Default] instance produces an empty schema.
+pub struct ContractV3 {
+    pub init:    Option<FunctionV3>,
+    pub receive: BTreeMap<String, FunctionV3>,
+}
+
+/// Contains event schema and associated event tag vector.
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct EventSchema {
+    /// Event schema.
+    pub event_schema: Type,
+    /// Event tags are mapped to enum indexes:
+    /// The event at index 0 in the event schema enum is mapped to the event tag
+    /// value `event_tags[0]`. The event at index 1 in the event schema enum
+    /// is mapped to the event tag value `event_tags[1]`. ...
+    pub event_tags:   Option<Vec<u8>>,
```

Comment:
I am not convinced it is a good idea to add these tags to the EventSchema.

In my view we should add a new constructor/variant to the `Schema::Type` enum that allows you to supply custom tags for your enum.

these tags can be either supplied via the derive annotation using an extra attribute, e.g., `tag =`, or via a manual implementation of `SchemaType` for the given type.

Would there be a problem with this?

We are in any case adding a new schema version.